tests/sys_atomic_utils/volatile_utils.h

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2020 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       For comparison: "Atomic" accesses using volatile
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * This header implements the `volatile_*()` family of functions of
```

Comment:
While these might not guarantee atomic access, they do guarantee volatile access (as in, don't optimize, don't re-use a previously read value, ...). Maybe that's useful on its own?

Reply:
Hmm. There might be use cases for that, but this is also provided by the `atomic_` functions (although potentially two instructions more expensive). It is often the case that one needs both the volatile access and the sequential order, but all current compilers move non-volatile code across volatile accesses.

I rather would not encourage programmer using this to safe 2 instructions, but risk bugs popping up due to incorrect use of them. But if you really want to export them, I could so and and I would add appropriate warnings. Hopefully people read those and think carefully before using them.

Reply:
> I rather would not encourage programmer using this to safe 2 instructions, but risk bugs popping up due to incorrect use of them.

ok!